File: Source/Entities/SampleSolid.vb
Imports Celeste.[Mod].Entities
Imports Microsoft.Xna.Framework

<CustomEntity("VeryBasicMod/SampleSolid")>
Public Class SampleSolid
    Inherits Solid
    Public Sub New(data As EntityData, offset As Vector2)
        MyBase.New(data.Position + offset, data.Width, data.Height, True)
        ' TODO: read properties from data
    End Sub

End Class

File: Source/Entities/SampleSolid.fs
namespace Celeste.Mod.VeryFunnyMod.Entities

open Celeste.Mod
open Celeste
open System
open Microsoft.Xna.Framework
open Celeste.Mod.Entities

[<CustomEntity("VeryFunnyMod/SampleSolid")>]
type SampleSolid(data: EntityData, offset: Vector2) as this =
    inherit Solid(data.Position + offset, float32 data.Width, float32 data.Height, true)
